### PR TITLE
Add Timestamp to PlainTextBackup

### LINKED ISSUE #5364: Enhancement: Append timestamp to export filename
I have:
- [X] searched open and closed issues for duplicates
- [X] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports
### Enhancement description

Right now, if you run a Plain Text Export, it exports to a file named `SignalPlaintextBackup.xml`

Since every time you backup, it uses the same filename, a user has to rename the file if they archive them off in a folder. It also makes it easy to accidentally overwrite an old backup when testing.

This request is to append a formatted timestamp to the filename, thus `SignalPlaintextBackup-YYYYMMDDHHSSmm.xml`, or `SignalPlaintextBackup-EPOCH.xml` or any other combination that would ensure unique filenames.

Suggest to include this with feature request: https://github.com/WhisperSystems/Signal-Android/issues/2791

Would likely require alterations here: https://github.com/WhisperSystems/Signal-Android/blob/3d651ef29a2b270ac291946ba6f0eda3db43fc6e/src/org/thoughtcrime/securesms/database/PlaintextBackupExporter.java#L15

And alterations to import code to select a file.
### Steps to reproduce

Just a plain text backup.

 **Signal version:** 3.14.2

### PR DESCRIPTION
**Pull request by char32**
### Contributor checklist

- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
  - Google Nexus 4 (Android 5.1.1)
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in the commit message of my first commit

---

Like requested in #5364 this patch adds a timestamp to the backup.
The file will be named like SignalPlainTextBackup_20160620201010.xml.
So yyyyMMddHHmmss as format.
To test it is just necessary to do a simple PlainTextBackup.

The import was adjusted too.
It behaves like this:

```
take backup-filename stored in environment
if not, search dynamically for the latest timestamp (based on filename) and use that backup
if not, search for the old filenmae SignalPlainTextBackup.xml (for backwards compatibilit
```

Resolves #5364

### CODE REVIEW

File: src/org/thoughtcrime/securesms/database/PlaintextBackupImporter.java [hunk 1]
```diff
@@ -34,10 +37,36 @@ private static void verifyExternalStorageForPlaintextImport() throws NoExternalS
...
+          return true;
+        } else {
+          return false;
+        }
+      }
+    };
+
+    File dir = Environment.getExternalStorageDirectory();
+    File[] allBackupFiles;
+    if(dir.exists()) {
```
Comment (haffenloher) [inline src/org/thoughtcrime/securesms/database/PlaintextBackupImporter.java]: i think there should be a space before the opening braces of your if conditions - at least that's how it's done in the rest of the project

File: src/org/thoughtcrime/securesms/database/PlaintextBackupImporter.java [hunk 2]
```diff
@@ -34,10 +37,36 @@ private static void verifyExternalStorageForPlaintextImport() throws NoExternalS
...
-    File backup    = PlaintextBackupExporter.getPlaintextExportFile();
-    File oldBackup = new File(Environment.getExternalStorageDirectory(), "TextSecurePlaintextBackup.xml");
+    File backup = PlaintextBackupExporter.getPlaintextExportFile();
+    FilenameFilter backupFilesFilter = new FilenameFilter() {
+      public boolean accept(File file, String name) {
+        if (name.startsWith("SignalPlaintextBackup_")) {
+          return true;
+        } else {
+          return false;
+        }
```
Comment (haffenloher) [inline src/org/thoughtcrime/securesms/database/PlaintextBackupImporter.java]: maybe replace this with just
`return name.startsWith("SignalPlaintextBackup_");`
?

**Comment by char32** [comment]
@haffenloher Thanks for the fast review! :-) I submitted the changes.
Btw. is it ok to have multipe commits or should I squash locally and upload the changes as one PR with single commit?

**Comment by haffenloher** [comment]
@char32 Usually it's better to preserve the commit history while your PR is still being reviewed :) Makes it easier to see what you changed.

File: src/org/thoughtcrime/securesms/database/PlaintextBackupImporter.java [hunk 3]
```diff
@@ -34,9 +37,31 @@ private static void verifyExternalStorageForPlaintextImport() throws NoExternalS
...
+    }
 
+    File oldBackup;
+    if (allBackupFiles.length > 0) {
+      oldBackup = allBackupFiles[0];
+    } else {
+      oldBackup = new File(Environment.getExternalStorageDirectory(), "SignalPlaintextBackup.xml");
+      if (!oldBackup.exists())
+        oldBackup = new File(Environment.getExternalStorageDirectory(), "TextSecurePlaintextBackup.xml");
+    }
```
Comment (FeuRenard) [inline src/org/thoughtcrime/securesms/database/PlaintextBackupImporter.java]: I don't know about our convention. In `ConversationActivity.java` you find two versions, both of them multiple times:

- https://github.com/WhisperSystems/Signal-Android/blob/master/src/org/thoughtcrime/securesms/ConversationActivity.java#L273
- https://github.com/WhisperSystems/Signal-Android/blob/master/src/org/thoughtcrime/securesms/ConversationActivity.java#L302

Comment (haffenloher) [inline src/org/thoughtcrime/securesms/database/PlaintextBackupImporter.java]: yep I think both of these are totally OK, but what about the third case where the body is too long to be in the same line? (like it is the case here)
I guess we could settle on using braces then, should be a little easier to read :) so +1 for your first comment

Comment (char32) [inline src/org/thoughtcrime/securesms/database/PlaintextBackupImporter.java]: True, it is easier to read :-)
Changed it accordingly

**Comment by nrizzio** [comment]
This pull request is obsolete because the plaintext backup is no longer offered and should be closed.

**Pull request closed without merging**